{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "25f35b18",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ImmGen.lab.contributed</th>\n",
       "      <th>sample.name</th>\n",
       "      <th>population.name</th>\n",
       "      <th>total.reads</th>\n",
       "      <th>overal_ alignment_rate%</th>\n",
       "      <th>mapped_MAPQ5</th>\n",
       "      <th>reads.after.removing.duplication</th>\n",
       "      <th>properly_paired.reads</th>\n",
       "      <th>paired.count</th>\n",
       "      <th>total.Htseq-count.on.genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Wagers</td>\n",
       "      <td>LTHSC.34-.BM.rep1</td>\n",
       "      <td>LTHSC.34-.BM</td>\n",
       "      <td>30846074</td>\n",
       "      <td>56,32</td>\n",
       "      <td>16524909</td>\n",
       "      <td>10069268</td>\n",
       "      <td>9582214</td>\n",
       "      <td>4791107</td>\n",
       "      <td>4641976</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Wagers</td>\n",
       "      <td>LTHSC.34-.BM.rep2</td>\n",
       "      <td>LTHSC.34-.BM</td>\n",
       "      <td>12221064</td>\n",
       "      <td>55,26</td>\n",
       "      <td>6432458</td>\n",
       "      <td>4424534</td>\n",
       "      <td>4103998</td>\n",
       "      <td>2051999</td>\n",
       "      <td>1984953</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Wagers</td>\n",
       "      <td>LTHSC.34+.BM.rep1</td>\n",
       "      <td>LTHSC.34+.BM</td>\n",
       "      <td>18001678</td>\n",
       "      <td>61,14</td>\n",
       "      <td>10460447</td>\n",
       "      <td>6606793</td>\n",
       "      <td>6263510</td>\n",
       "      <td>3131755</td>\n",
       "      <td>3030056</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Wagers</td>\n",
       "      <td>LTHSC.34+.BM.rep2</td>\n",
       "      <td>LTHSC.34+.BM</td>\n",
       "      <td>20921592</td>\n",
       "      <td>61,41</td>\n",
       "      <td>12255655</td>\n",
       "      <td>8074178</td>\n",
       "      <td>7652366</td>\n",
       "      <td>3826183</td>\n",
       "      <td>3709788</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Wagers</td>\n",
       "      <td>STHSC.150-.BM.rep1</td>\n",
       "      <td>STHSC.150-.BM</td>\n",
       "      <td>26562826</td>\n",
       "      <td>55,4</td>\n",
       "      <td>14008223</td>\n",
       "      <td>8315375</td>\n",
       "      <td>7899112</td>\n",
       "      <td>3949556</td>\n",
       "      <td>3824397</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  ImmGen.lab.contributed         sample.name population.name  total.reads  \\\n",
       "0                 Wagers   LTHSC.34-.BM.rep1    LTHSC.34-.BM     30846074   \n",
       "1                 Wagers   LTHSC.34-.BM.rep2    LTHSC.34-.BM     12221064   \n",
       "2                 Wagers   LTHSC.34+.BM.rep1    LTHSC.34+.BM     18001678   \n",
       "3                 Wagers   LTHSC.34+.BM.rep2    LTHSC.34+.BM     20921592   \n",
       "4                 Wagers  STHSC.150-.BM.rep1   STHSC.150-.BM     26562826   \n",
       "\n",
       "  overal_ alignment_rate%  mapped_MAPQ5  reads.after.removing.duplication  \\\n",
       "0                   56,32      16524909                          10069268   \n",
       "1                   55,26       6432458                           4424534   \n",
       "2                   61,14      10460447                           6606793   \n",
       "3                   61,41      12255655                           8074178   \n",
       "4                    55,4      14008223                           8315375   \n",
       "\n",
       "   properly_paired.reads  paired.count  total.Htseq-count.on.genes  \n",
       "0                9582214       4791107                     4641976  \n",
       "1                4103998       2051999                     1984953  \n",
       "2                6263510       3131755                     3030056  \n",
       "3                7652366       3826183                     3709788  \n",
       "4                7899112       3949556                     3824397  "
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "ATAC = pd.read_csv(\"data/1.ATAC-Seq data.CSV\", sep=\";\")\n",
    "QC_RD = pd.read_csv(\"data/ATAC_QCmatric_ReadStatistics.csv\", sep=\";\")\n",
    "QC_RD.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "6a674f58",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 169 entries, 0 to 168\n",
      "Data columns (total 10 columns):\n",
      " #   Column                            Non-Null Count  Dtype \n",
      "---  ------                            --------------  ----- \n",
      " 0   ImmGen.lab.contributed            169 non-null    object\n",
      " 1   sample.name                       169 non-null    object\n",
      " 2   population.name                   169 non-null    object\n",
      " 3   total.reads                       169 non-null    int64 \n",
      " 4   overal_ alignment_rate%           169 non-null    object\n",
      " 5   mapped_MAPQ5                      169 non-null    int64 \n",
      " 6   reads.after.removing.duplication  169 non-null    int64 \n",
      " 7   properly_paired.reads             169 non-null    int64 \n",
      " 8   paired.count                      169 non-null    int64 \n",
      " 9   total.Htseq-count.on.genes        169 non-null    int64 \n",
      "dtypes: int64(6), object(4)\n",
      "memory usage: 13.3+ KB\n"
     ]
    }
   ],
   "source": [
    "QC_RD.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "6683497f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(169, 10)"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "QC_RD.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "39f033fb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ImmGen.lab.contributed</th>\n",
       "      <th>sample.name</th>\n",
       "      <th>population.name</th>\n",
       "      <th>total.reads</th>\n",
       "      <th>overal_ alignment_rate%</th>\n",
       "      <th>mapped_MAPQ5</th>\n",
       "      <th>reads.after.removing.duplication</th>\n",
       "      <th>properly_paired.reads</th>\n",
       "      <th>paired.count</th>\n",
       "      <th>total.Htseq-count.on.genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>169</td>\n",
       "      <td>169</td>\n",
       "      <td>169</td>\n",
       "      <td>1.690000e+02</td>\n",
       "      <td>169</td>\n",
       "      <td>1.690000e+02</td>\n",
       "      <td>1.690000e+02</td>\n",
       "      <td>1.690000e+02</td>\n",
       "      <td>1.690000e+02</td>\n",
       "      <td>1.690000e+02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>unique</th>\n",
       "      <td>13</td>\n",
       "      <td>169</td>\n",
       "      <td>86</td>\n",
       "      <td>NaN</td>\n",
       "      <td>161</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>top</th>\n",
       "      <td>CBDM</td>\n",
       "      <td>LTHSC.34-.BM.rep1</td>\n",
       "      <td>B.Fo.Sp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>52,41</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>freq</th>\n",
       "      <td>45</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.055347e+07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.082893e+07</td>\n",
       "      <td>6.803538e+06</td>\n",
       "      <td>6.457130e+06</td>\n",
       "      <td>3.228565e+06</td>\n",
       "      <td>3.130647e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.156571e+06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.937772e+06</td>\n",
       "      <td>2.206872e+06</td>\n",
       "      <td>2.141713e+06</td>\n",
       "      <td>1.070856e+06</td>\n",
       "      <td>1.040051e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.762368e+06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.145012e+06</td>\n",
       "      <td>2.438597e+06</td>\n",
       "      <td>2.283692e+06</td>\n",
       "      <td>1.141846e+06</td>\n",
       "      <td>1.108508e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.470354e+07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.019663e+06</td>\n",
       "      <td>5.332785e+06</td>\n",
       "      <td>5.017788e+06</td>\n",
       "      <td>2.508894e+06</td>\n",
       "      <td>2.446066e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.083229e+07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.028266e+07</td>\n",
       "      <td>6.466320e+06</td>\n",
       "      <td>6.119962e+06</td>\n",
       "      <td>3.059981e+06</td>\n",
       "      <td>2.962653e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.630127e+07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.377734e+07</td>\n",
       "      <td>8.405751e+06</td>\n",
       "      <td>7.987896e+06</td>\n",
       "      <td>3.993948e+06</td>\n",
       "      <td>3.876989e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.062597e+07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.284949e+07</td>\n",
       "      <td>1.972867e+07</td>\n",
       "      <td>1.918715e+07</td>\n",
       "      <td>9.593576e+06</td>\n",
       "      <td>9.330719e+06</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       ImmGen.lab.contributed        sample.name population.name  \\\n",
       "count                     169                169             169   \n",
       "unique                     13                169              86   \n",
       "top                      CBDM  LTHSC.34-.BM.rep1         B.Fo.Sp   \n",
       "freq                       45                  1               4   \n",
       "mean                      NaN                NaN             NaN   \n",
       "std                       NaN                NaN             NaN   \n",
       "min                       NaN                NaN             NaN   \n",
       "25%                       NaN                NaN             NaN   \n",
       "50%                       NaN                NaN             NaN   \n",
       "75%                       NaN                NaN             NaN   \n",
       "max                       NaN                NaN             NaN   \n",
       "\n",
       "         total.reads overal_ alignment_rate%  mapped_MAPQ5  \\\n",
       "count   1.690000e+02                     169  1.690000e+02   \n",
       "unique           NaN                     161           NaN   \n",
       "top              NaN                   52,41           NaN   \n",
       "freq             NaN                       3           NaN   \n",
       "mean    2.055347e+07                     NaN  1.082893e+07   \n",
       "std     7.156571e+06                     NaN  3.937772e+06   \n",
       "min     6.762368e+06                     NaN  3.145012e+06   \n",
       "25%     1.470354e+07                     NaN  8.019663e+06   \n",
       "50%     2.083229e+07                     NaN  1.028266e+07   \n",
       "75%     2.630127e+07                     NaN  1.377734e+07   \n",
       "max     5.062597e+07                     NaN  3.284949e+07   \n",
       "\n",
       "        reads.after.removing.duplication  properly_paired.reads  paired.count  \\\n",
       "count                       1.690000e+02           1.690000e+02  1.690000e+02   \n",
       "unique                               NaN                    NaN           NaN   \n",
       "top                                  NaN                    NaN           NaN   \n",
       "freq                                 NaN                    NaN           NaN   \n",
       "mean                        6.803538e+06           6.457130e+06  3.228565e+06   \n",
       "std                         2.206872e+06           2.141713e+06  1.070856e+06   \n",
       "min                         2.438597e+06           2.283692e+06  1.141846e+06   \n",
       "25%                         5.332785e+06           5.017788e+06  2.508894e+06   \n",
       "50%                         6.466320e+06           6.119962e+06  3.059981e+06   \n",
       "75%                         8.405751e+06           7.987896e+06  3.993948e+06   \n",
       "max                         1.972867e+07           1.918715e+07  9.593576e+06   \n",
       "\n",
       "        total.Htseq-count.on.genes  \n",
       "count                 1.690000e+02  \n",
       "unique                         NaN  \n",
       "top                            NaN  \n",
       "freq                           NaN  \n",
       "mean                  3.130647e+06  \n",
       "std                   1.040051e+06  \n",
       "min                   1.108508e+06  \n",
       "25%                   2.446066e+06  \n",
       "50%                   2.962653e+06  \n",
       "75%                   3.876989e+06  \n",
       "max                   9.330719e+06  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "QC_RD.describe(include='all')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "envBCells1",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
